hare/hare.go

```diff
@@ -160,21 +160,23 @@ var ErrTooLate = errors.New("consensus process finished too late")
 
 // records the provided output.
 func (h *Hare) collectOutput(ctx context.Context, output TerminationOutput) error {
-	set := output.Set()
-	blocks := make([]types.BlockID, len(set.values))
-	i := 0
-	for v := range set.values {
-		blocks[i] = v
-		i++
-	}
+	var blocks []types.BlockID
+	if output.Completed() {
```

Comment:
This seems redundant to me.

If `output.Completed() == false` then `len(output.Set()) == 0` and then the loop runs zero times and with an empty blocks list `h.validate(blocks)` does nothing and returns `true`.